webhooks-extension/endpoints/sink.go

```diff
@@ -0,0 +1,342 @@
+
+	log.Printf("GitHub event type is %s", gitHubEventTypeString)
+
+	timestamp := getDateTimeAsString()
+
+	if gitHubEventTypeString == "ping" {
+		response.WriteHeader(http.StatusNoContent)
+	} else if gitHubEventTypeString == "push" {
+		log.Print("Handling a push event...")
+
+		webhookData := gh.PushPayload{}
+
+		if err := request.ReadEntity(&webhookData); err != nil {
+			log.Printf("an error occurred decoding webhook data: %s", err)
+			return
+		}
+
+		buildInformation.REPOURL = webhookData.Repository.URL
+		buildInformation.SHORTID = webhookData.HeadCommit.ID[0:7]
+		buildInformation.COMMITID = webhookData.HeadCommit.ID
+		buildInformation.REPONAME = webhookData.Repository.Name
+		buildInformation.TIMESTAMP = timestamp
+
+		createPipelineRunFromWebhookData(buildInformation, r)
+		log.Printf("Build information for repository %s:%s %s", buildInformation.REPOURL, buildInformation.SHORTID, buildInformation)
+
+	} else if gitHubEventTypeString == "pull_request" {
+		log.Print("Handling a pull request event...")
+
+		webhookData := gh.PullRequestPayload{}
+
+		if err := request.ReadEntity(&webhookData); err != nil {
+			log.Printf("an error occurred decoding webhook data: %s", err)
+			return
+		}
+
+		buildInformation.REPOURL = webhookData.Repository.HTMLURL
+		buildInformation.SHORTID = webhookData.PullRequest.Head.Sha[0:7]
+		buildInformation.COMMITID = webhookData.PullRequest.Head.Sha
+		buildInformation.REPONAME = webhookData.Repository.Name
+		buildInformation.TIMESTAMP = timestamp
+
+		createPipelineRunFromWebhookData(buildInformation, r)
+		log.Printf("Build information for repository %s:%s %s", buildInformation.REPOURL, buildInformation.SHORTID, buildInformation)
+
+	} else {
+		log.Print("event wasn't a push or pull event, no action will be taken")
+	}
+}
+
+// This is the main flow that handles building and deploying: given everything we need to kick off a build, do so
+func createPipelineRunFromWebhookData(buildInformation BuildInformation, r Resource) {
+	log.Printf("In createPipelineRunFromWebhookData, build information: %s", buildInformation)
+
+	// TODO: Use the dashboard endpoint to create the PipelineRun
+	// Track PR: https://github.com/tektoncd/dashboard/pull/33
+	// and issue: https://github.com/tektoncd/dashboard/issues/47
+
+	// These can be set either when creating the event handler/github source manually through yml or when installing the Helm chart.
+	// For the chart, PIPELINE_RUN_NAMESPACE picks up the specified namespace. If this is set it will be used.
+
+	// Otherwise, we use the namespace where this has been installed. This allows us to have PipelineRuns in namespaces other than the installed to namespace
+	// but may require additional RBAC configuration depending on your cluster configuration and service account permissions.
+
+	// The specified service account name is also exposed through the chart's values.yaml: defaulting to "tekton-pipelines".
+
```

Comment:
Lots of comments above to env vars from chart ..... might need tidying up as no helm chart here ... also RBAC .... might need to put a note about rbac in readme